persistence/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/AbstractPersistentEntityRegistry.scala

```diff
@@ -72,7 +72,9 @@ class AbstractPersistentEntityRegistry(system: ActorSystem) extends PersistentEn
   }
 
   private val extractShardId: ShardRegion.ExtractShardId = {
-    case CommandEnvelope(entityId, payload) =>
+    case CommandEnvelope(entityId, _) =>
+      (math.abs(entityId.hashCode) % maxNumberOfShards).toString
```

Comment:
this is in 4 places now, would be good to extract this to one shared utility function

Reply:
Good call!

Reply:
Where do you suggest a good place for that shared utility would be?

Reply:
A new utility `object HashCodeMessageExtractor` in persistence/core/src/main/scala/com/lightbend/lagom/internal/persistence/cluster/

Reply:
Done.

Reply:
@patriknw Just an FYI, this is ready for a final review :)